pmd-java/src/main/resources/category/java/errorprone.xml

```diff
@@ -3127,12 +3127,13 @@ public boolean test(String s) {
     <rule name="UselessOperationOnImmutable"
           language="java"
           since="3.5"
-          message="An operation on an Immutable object (String, BigDecimal or BigInteger) won't change the object itself"
+          message="An operation on an Immutable object (String, BigDecimal, BigInteger or java.time.*) won't change the object itself"
```

Comment:
Maybe it makes sense to not list these types here, they're already in the description of the rule.
```suggestion
          message="An operation on an immutable object won't change the object itself"
```
Or maybe we should just be more descriptive as to what the rule flags:
```suggestion
          message="The result of an operation on an immutable object is ignored"
```
The developer's intention (whether they thought they would modify the object or not) is irrelevant. It's also possible that this is a copy paste error or another kind of mistake

Reply:
Yes, I like your second suggestion best.